calicoctl/tests/unit/pool_test.py

```diff
@@ -102,3 +101,40 @@ def test_ip_pool_remove(self, m_client, m_sleep):
         self.assertEqual(m_pool.disabled, True)
         m_client.release_pool_affinities.assert_called_once_with(m_pool)
         m_client.remove_ip_pool.assert_called_once_with(4, net1.ip)
+
+    @patch("calico_ctl.pool.client", autospec=True)
+    def test_add_overlapping_existing_pool(self, m_client):
+        """
+        Test ip_pool_add exits when a pool is added that overlaps with an
+        existing pool.
+        """
+        # Test adding pool that falls within an existing pool
+        m_client.get_ip_pools = Mock()
+        m_client.get_ip_pools.return_value = [IPPool("10.10.10.0/24")]
+        with patch('sys.exit', autospec=True) as m_sys_exit:
+            # Call method under test
+            pool.ip_pool_add(["10.10.10.0/25"], 4, False, False)
```

Comment:
Would be more clear for the next guy if you key value'd these params when calling